lib/Conversion/StandardToHandshake/StandardToHandshake.cpp

```diff
@@ -804,20 +1206,6 @@ LogicalResult checkMergePredecessors(MergeLikeOpInterface mergeOp) {
              << getBlockPredecessorCount(block) << " predecessor blocks";
   }
 
-  // There must be a predecessor from each predecessor block
```

Comment:
I see this no longer holds for all merge-like ops, but are there cases where we should still enforce this on some ops?

Reply:
Hmm... This commit uses all three merge-like operations in a way where they will only have inputs from a subset of the predecessors. The check could be enabled if we carry in information about what we detected as loop headers (And therefore, rewrote) and exclude these from the check.

Reply:
I see. Well it seems like a pretty big relaxing to the verifier for anything that isn't a loop header. If it's not a huge pain, being able to keep this check and just exclude loop headers seems valuable.